src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,80 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  addComment: (newComment: CommentData) => void;
+  buttonAddLoading: boolean;
+  selectedPost: Post | null;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  addComment,
+  buttonAddLoading,
+  selectedPost,
+}) => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [bodyComment, setBodyComment] = useState('');
+  const [bodyCommentError, setBodyCommentError] = useState(false);
+
+  let newComment: CommentData;
+
+  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setNameError(false);
+    setEmailError(false);
+    setBodyCommentError(false);
+
+    let hasError = false;
+
+    if (name.trim().length === 0) {
+      setNameError(true);
+      hasError = true;
+    }
+
+    if (email.trim().length === 0) {
+      setEmailError(true);
+      hasError = true;
+    }
+
+    if (bodyComment.trim().length === 0) {
+      setBodyCommentError(true);
+      hasError = true;
+    }
```

Comment:
You're checking if the fields are empty one by one. It might be better to create an array of fields and iterate over it to check each one. This would make your code more DRY (Don't Repeat Yourself).